### PR TITLE
Scripts to perform edge detection with Sobel filter

### PR DESCRIPTION
**Pull request by Namyalg**
This script performs edge detection using the Sobel filter, (no inbuilt functions from the openCV library are used)

Issue relates to #93

### CODE REVIEW

File: Image-Processing/Sobel-edge-detection/sobel-edge-detect.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import cv2
+from google.colab.patches import cv2_imshow
+from math import sqrt, atan
+
+#This script can be used for edge-detection without the use of any inbuilt openCV libraries
+#Here the sobel filter is applied after blurring the image.
+
+kernelblur = np.array([1/16,1/8,1/16,1/8,1/4,1/8,1/16,1/8,1/16]).reshape(3,3)
```
Comment (disha2sinha) [inline Image-Processing/Sobel-edge-detection/sobel-edge-detect.py]: put space after comma

File: Image-Processing/Sobel-edge-detection/sobel-edge-detect.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import cv2
+from google.colab.patches import cv2_imshow
+from math import sqrt, atan
+
+#This script can be used for edge-detection without the use of any inbuilt openCV libraries
+#Here the sobel filter is applied after blurring the image.
+
+kernelblur = np.array([1/16,1/8,1/16,1/8,1/4,1/8,1/16,1/8,1/16]).reshape(3,3)
+kernelsobelX = np.array([-1,-2,-1,0,0,0,1,2,1]).reshape(3,3)
```
Comment (disha2sinha) [inline Image-Processing/Sobel-edge-detection/sobel-edge-detect.py]: Space after comma

File: Image-Processing/Sobel-edge-detection/sobel-edge-detect.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      product = np.multiply(temp, kernel)
+      apply_filter.append(sum(sum(product)))
+
+  #The pixels are converted back to the image
+  apply_filter = np.array(apply_filter).reshape(height-1, width-1)
+  return(apply_filter)
+
+def sobel(img):
+  height = img.shape[0]
+  width= img.shape[1]
```
Comment (disha2sinha) [inline Image-Processing/Sobel-edge-detection/sobel-edge-detect.py]: insert space before comma

File: Image-Processing/Sobel-edge-detection/sobel-edge-detect.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  #The sobel filter is applied in the X and Y direction separately
+  convoluted_Y = convolute(blur, kernelsobelX, height, width)
+  convoluted_X = convolute(blur, kernelsobelY, height, width)
+
+  #Every pixel in convoluted_X can be called gx, in convoluted_Y can be called gy,
+  #The resultant will be sqrt(pow(gx**2) + pow(gy**2))
+  resultant = []
+  for i in range(height-1):
+    for j in range(width-1):
+      in_x = pow(convoluted_X[i,j] ,2)
```
Comment (disha2sinha) [inline Image-Processing/Sobel-edge-detection/sobel-edge-detect.py]: Give space after comma and remove space before comma

File: Image-Processing/Sobel-edge-detection/sobel-edge-detect.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  
+  #The sobel filter is applied in the X and Y direction separately
+  convoluted_Y = convolute(blur, kernelsobelX, height, width)
+  convoluted_X = convolute(blur, kernelsobelY, height, width)
+
+  #Every pixel in convoluted_X can be called gx, in convoluted_Y can be called gy,
+  #The resultant will be sqrt(pow(gx**2) + pow(gy**2))
+  resultant = []
+  for i in range(height-1):
+    for j in range(width-1):
```
Comment (disha2sinha) [inline Image-Processing/Sobel-edge-detection/sobel-edge-detect.py]: insert space around -

File: Image-Processing/Sobel-edge-detection/sobel-edge-detect.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      in_x = pow(convoluted_X[i,j] ,2)
+      in_y = pow(convoluted_Y[i,j] , 2)
+      gradient = sqrt(in_x + in_y)
+      reusultant.append(grad) 
+  resultant = np.array(resultant).reshape(height-1, width-1)
+  cv2_imshow(resultant)
+
+if __name__ == "__main__":
```
Comment (disha2sinha) [inline Image-Processing/Sobel-edge-detection/sobel-edge-detect.py:74]: Space around ==

**Review by disha2sinha — APPROVED** [review]

**Comment by Namyalg** [comment]
Okay will make the changes immediately

**Review by disha2sinha — CHANGES_REQUESTED** [review]
Make the changes stated below and you are good to go

**Comment by Namyalg** [comment]
Yep, all done

**Review by disha2sinha — APPROVED** [review]

**Pull request merged**